### PR TITLE
Adds Prayer Action Menu

### PR DESCRIPTION
**Pull request by cdanwards**
## DESCRIPTION

As a User, I want to be able to interact with the prayers that I'm praying for in the app.

### What does this PR do, or why is it needed?

This PR address a few things:
It introduces the `PrayerActionMenu` which allows a user to 1) Exit Prayer, 2) Save a Prayer or 3) Scroll to the next prayer.
This alters the `PrayerCard` to include the `PrayerActionMenu`. We had tried to just render it in the `PrayerList` but the state of all items was being changed, not the state of the individual card.
This also brings in a `FlatList` in the Prayer List in favor of a `FeedView` because of concerns addressed [here](https://github.com/ApollosProject/apollos-prototype/issues/673)

### How do I test this PR?

1. Navigate to Prayer
2. Select a prayer feed
3. Select a prayer.
4. Test the different press options presented. Exit will navigate to the Prayer tab. Saving will save the prayer to your saved Prayer List. Selected next prayer will scroll to the next prayer in the list.

---

> I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))

## TODO

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [DEV-XXX](#url)
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android if applicable
- [ ] Set two relevant reviewers

## REVIEW

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCard.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}))(PaddedView);
+
+const StyledTouchable = styled(({ theme }) => ({
+  alignItems: 'center',
+  padding: theme.sizing.baseUnit * 1.5,
+}))(Touchable);
+
+const PrayerActionMenuCard = memo(
+  ({ exitPrayer, savePrayer, advancePrayer, expandedHeight }) => (
+    <>
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCard.js]: Why is this needed?

File: packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCard.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      <StyledCard expandedHeight={expandedHeight}>
+        <StyledCardContent>
+          <StyledPaddedView>
+            <StyledTouchable onPress={() => exitPrayer()}>
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCard.js:53]:
```suggestion
            // TODO: Practice good DRY methods by combing these three touchables into a reusable component
            <StyledTouchable onPress={() => exitPrayer()}>
```

File: packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCard.js (hunk 1 above)
Comment (cdanwards) [reply packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCard.js]: Took them out. They were there from a previous version of this file.

File: packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCard.js (hunk 2 above)
Comment (cdanwards) [reply packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCard.js:53]: Left this comment in the code.

**Comment by IsaacHardy** [comment]
Mutation tries to trigger on clicking a saved prayer in the saved prayuer list
Simulator Screen Shot - iPhone X - 2019-05-02 at 18 06 21

**Comment by IsaacHardy** [comment]
Tries to increment prayer on the "My Prayer" cards, too.

File: packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCardConnected.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const PrayerActionMenuCardConnected = memo(
+  ({ onAdvancePrayer, navigation, prayerId, ...props }) => {
+    const exitPrayer = () => navigation.navigate('Prayer');
+    return (
+      <Mutation mutation={savePrayerRequest}>
+        {(savePrayer) => (
+          <PrayerActionMenuCard
+            exitPrayer={exitPrayer}
+            savePrayerId={async () => {
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/prayer/PrayerActionMenuCard/PrayerActionMenuCardConnected.js]:
```suggestion
            savePrayer={async () => {
```

**Review by IsaacHardy — DISMISSED** [review]

**Review by IsaacHardy — APPROVED** [review]
